apps/testing-javascript/src/path-to-purchase-react/use-coupon.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+// import RedeemDialog from './redeem-dialog'
+
+type CouponValidator = {isValid: boolean; id: string; isRedeemable: boolean}
+
+export function useCoupon(coupon?: CouponValidator) {
+  const [validCoupon, setValidCoupon] = React.useState(false)
+  React.useEffect(() => {
+    setTimeout(() => {
+      setValidCoupon(Boolean(coupon && coupon.isValid))
+    }, 0)
+  }, [])
+
+  return {
+    validCoupon,
+    redeemableCoupon: validCoupon && coupon?.isRedeemable,
+    RedeemDialogForCoupon: () => {
+      return coupon ? (
+        // <RedeemDialog open={validCoupon} couponId={coupon.id} />
```

Comment:
Does this dialog need to be restored before this gets merged?

Reply:
there is a lot of stuff to clean up still, will do it in next PR, decided to leave the blocks I was not sure do we need it or not for now.